hub/api/pytorch.py

```diff
@@ -0,0 +1,241 @@
+import os
+import pickle
+import numpy as np
+from itertools import repeat
+from collections import defaultdict
+from hub.util.keys import get_index_map_key
+from hub.core.chunk_engine.chunker import join_chunks
+from hub.core.chunk_engine.read import read_tensor_meta
+from pathos.pools import ProcessPool
+from multiprocessing import resource_tracker
+from multiprocessing.shared_memory import SharedMemory
+from hub.util.exceptions import ModuleNotInstalledException
+
+
+# TODO make this use shared memory to make on the fly transforms faster. Currently using transform slows us down by 10x
+def apply_transform(transform, sample):
+    """Used to apply transforms to a single samples"""
+    return transform(sample) if transform else sample
+
+
+def shared_memory_clear(chunk_set):
+    """Helper function that checks if an existing SharedMemory exists for any chunk in chunk_set and clears it"""
+    for chunk_name in chunk_set:
+        try:
+            shm = SharedMemory(name=chunk_name)
+            shm.close()
+            shm.unlink()
+        except:
+            pass
+
+
+def read_chunk(chunk, key):
+    """Reads a single chunk from the dataset's storage provider and stores it in the SharedMemory"""
```

Comment:
`read_chunk` isn't all this method is doing, according to your docstring it's also storing in shared memory. can you rename this method to represent this?